collect_app/src/main/java/org/odk/collect/android/instancemanagement/send/ReadyToSendViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package org.odk.collect.android.instancemanagement.send
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import org.odk.collect.async.Scheduler
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.forms.instances.InstancesRepository
+import java.util.function.Supplier
+
+class ReadyToSendViewModel(
+    private val instancesRepository: InstancesRepository,
+    private val scheduler: Scheduler,
+    private val clock: Supplier<Long>
+) : ViewModel() {
+    private val _data = MutableLiveData<Data>()
+    val data: LiveData<Data> = _data
+
+    fun init() {
```

Comment:
I think we're good to just have this executed when the ViewModel is instantiated rather than needing it called explicitly (like we do in `BlankFormListViewModel`. ViewModel caching handles optimising that for us.

Reply:
I needed this to update the banner after sending forms. For example, initially, it might be invisible because there are forms ready to send but not sent forms and when you send one it should become visible. It might be not needed if we are going to keep the change you introduced in https://github.com/getodk/collect/pull/5717 - finishing the activity after sending forms and navigating to the main menu. Will this be merged into master?

Reply:
Yeah it will be! I'm going to make a PR for that today, so we won't need `init`.

Reply:
Actually, I've got myself confused here. We're not going to add the changes from #5717 to `master` - we're going to include them in v2023.3, not v2023.4. Also, the changes I think you're referring to is that we now return to the main menu after viewing or editing a form from "Ready to send", but we still return to "Read to send" after sending forms. That being the case, you will need to update the view model again when returning (until we have a similar setup for instances that we do for forms with `FormsDataService`). I guess that's easy enough to add by creating an `update` method that the view model's `init` can call. I'm guessing this will be the first thing that @getodk/testers will find now that it's marked "needs testing" 😆.

Reply:
Right, fixed.